qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -268,23 +294,33 @@ clone_image=test_clone
 clone_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} ${clone_image}
 write_image ${CLUSTER2} ${POOL} ${clone_image} 100
 
-enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} journal
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  enable_mirror ${CLUSTER2} ${POOL} ${clone_image} snapshot
+else
+  enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} journal
+fi
 wait_for_image_replay_started ${CLUSTER1} ${PARENT_POOL} ${parent_image}
 wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${PARENT_POOL} ${parent_image}
-wait_for_status_in_pool_dir ${CLUSTER1} ${PARENT_POOL} ${parent_image} 'up+replaying' 'primary_position'
+wait_for_replay_status_in_pool_dir ${CLUSTER1} ${PARENT_POOL} ${parent_image}
 compare_images ${PARENT_POOL} ${parent_image}
 
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${clone_image}
 wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${clone_image}
-wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${clone_image} 'up+replaying' 'primary_position'
+wait_for_replay_status_in_pool_dir ${CLUSTER1} ${POOL} ${clone_image}
 compare_images ${POOL} ${clone_image}
 remove_image_retry ${CLUSTER2} ${POOL} ${clone_image}
 
 testlog " - clone v1"
-clone_image ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
-
-clone_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} \
-            ${clone_image}_v1 --rbd-default-clone-format 1
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  clone_image_and_enable_mirror ${CLUSTER1} ${PARENT_POOL} ${parent_image} \
+	${parent_snap} ${POOL} ${clone_image}1
+  clone_image_and_enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} \
+	${parent_snap} ${POOL} ${clone_image}_v1 snapshot --rbd-default-clone-format 1
+else
+  clone_image ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
+  clone_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} \
+	${clone_image}_v1 --rbd-default-clone-format 1
```

Comment:
It seems like we could just have `clone_image` function that would do `enable_mirror` if the mirror mode snapshot. Then we could have less these `if [ "${RBD_MIRROR_MODE}" = "snapshot" ]` conditions in the test itself. But I don't have a good opinion about this, if it is a really good idea.

Reply:
clone_image is still needed for snapshot test cases which doesn't need enable mirror, hence introduced a new function. This is taken care now.